hazelcast-cloud/src/main/java/com/hazelcast/aws/impl/DescribeInstances.java

```diff
@@ -59,6 +66,29 @@ public DescribeInstances(AwsConfig awsConfig, String endpoint) {
         attributes.put("X-Amz-Expires", "30");
     }
 
+    private void getKeysFromIamRole() {
+        try {
+            String query = "latest/meta-data/iam/security-credentials/" + awsConfig.getIamRole();
+            URL url = new URL("http", "169.254.169.254", -1, query);
+            InputStreamReader is = new InputStreamReader(url.openStream(), "UTF-8");
+            BufferedReader reader = new BufferedReader(is);
+            Pattern pattern = Pattern.compile(": \"(.*?)\",");
+            for (String line; (line = reader.readLine()) != null; ) {

```

Comment:
I am sure you can create a static util method which parses response string and returns a map which contains required parameters. That makes testing this part easier...